src/button/button.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * JavaScript 'shim' to trigger the click event of element(s) when the space key is pressed.
+ *
+ * Created since some Assistive Technologies (for example some Screenreaders)
+ * will tell a user to press space on a 'button', so this functionality needs to be shimmed
+ * See https://github.com/alphagov/govuk_elements/pull/272#issuecomment-233028270
+ *
+ * Usage instructions:
+ * the 'shim' will be automatically initialised
+ */
+;(function (global) {
+  'use strict'
+
+  var GOVUK_FRONTEND = global.GOVUK_FRONTEND || {}
+  var KEY_SPACE = 32
+
+  GOVUK_FRONTEND.buttons = {
+
+    /**
+    * Add event construct for modern browsers or IE
```

Comment:
Anywhere we are polyfilling older browsers can we list the browser support?

This will help future people remove code if they know a browser is no longer needed for example.

Reply:
jQuery does this like so

```js
// Support: IE 11
if (bunchOfCodeHere) {}
```

Reply:
updated as suggested